### PR TITLE
New package list service

### PR DESCRIPTION
**Pull request by kamildoleglo**
The name `guessLocation` is open for discussion

Last commit is just an integration test

### CODE REVIEW

**Review by BarkingBad — APPROVED** [review]

File: plugins/base/src/main/kotlin/renderers/PackageListService.kt [hunk 1]
```diff
@@ -2,44 +2,43 @@ package org.jetbrains.dokka.base.renderers
...
-            contentPage?.dri?.forEach {
-                if (parentDris.isNotEmpty() && it.parent !in parentDris) {
-                    locationProvider.resolve(node)
-                        ?.let { nodeLocation -> nonStandardLocations[it.toString()] = nodeLocation }
-                        ?: context.logger.error("Cannot resolve path for ${node.name}!")
+            contentPage?.dri?.forEach { dri ->
+                val nodeLocation = locationProvider.resolve(node, context = module, skipExtension = true)
+                    ?: run { context.logger.error("Cannot resolve path for ${node.name}!"); null }
+
+                if (locationProvider.guessLocation(dri) != nodeLocation && dri != DRI.topLevel) {
```
Comment (sellophane) [inline plugins/base/src/main/kotlin/renderers/PackageListService.kt:33]: Please swap the conditions here, `&&` is short-circuit  and there is no need to guess location if dri is top-level

File: plugins/base/src/main/kotlin/resolvers/local/LocationProvider.kt [hunk 2]
```diff
@@ -10,6 +11,18 @@ interface LocationProvider {
...
     fun pathToRoot(from: PageNode): String
     fun ancestors(node: PageNode): List<PageNode>
+
+    /**
+     * This method should return guessed filesystem location for a given [DRI]
+     * It is used to decide if a [DRI] should be present in the relocation list of the
+     * generated package-list so it is ok if the path differs from the one returned by [resolve]
+     * @return Path to a giver [DRI] or null if path should not be considered for relocations
+     */
+    fun guessLocation(dri: DRI): String =
```
Comment (sellophane) [inline plugins/base/src/main/kotlin/resolvers/local/LocationProvider.kt:21]: Something like `createDefaultLocationForDri`maybe?

**Review by sellophane — CHANGES_REQUESTED** [review]

File: plugins/base/src/main/kotlin/resolvers/local/LocationProvider.kt (hunk 2 above)
Comment (kamildoleglo) [reply plugins/base/src/main/kotlin/resolvers/local/LocationProvider.kt]: Changed

File: plugins/base/src/main/kotlin/renderers/PackageListService.kt (hunk 1 above)
Comment (kamildoleglo) [reply plugins/base/src/main/kotlin/renderers/PackageListService.kt]: Fixed

File: integration-tests/gradle/projects/it-multimodule-1/gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.1-bin.zip
```
[the remark this answers is not in the document]
Comment (kamildoleglo) [reply integration-tests/gradle/projects/it-multimodule-1/gradle/wrapper/gradle-wrapper.properties:3]: Not making much of a difference tbh, version is set by the test runner itself, this is only if one would want to build this project manually

**Review by sellophane — APPROVED** [review]

**Pull request merged**